library/core/src/ffi/c_str.rs

```diff
@@ -885,7 +885,32 @@ mod verify {
             assert!(c_str.is_safe());
         }
     }
-  
+
+    // pub const fn from_bytes_with_nul(bytes: &[u8]) -> Result<&Self, FromBytesWithNulError>
+    #[kani::proof]
+    #[kani::unwind(17)]  
+    fn check_from_bytes_with_nul() {
+        const MAX_SIZE: usize = 16;
+        let mut string: [u8; MAX_SIZE] = kani::any();
+        let null_pos: usize = kani::any_where(|x| *x < MAX_SIZE);
```

Comment:
you could use `any_slice` instead.

Reply:
sure, that was our initial approach as well, something like : 
```
    #[kani::proof]
    #[kani::unwind(17)] 
    fn check_from_bytes_with_nul2() {
        const MAX_SIZE: usize = 16;
        let string: [u8; MAX_SIZE] = kani::any();
        let slice = kani::slice::any_slice_of_array(&string);

        let result = CStr::from_bytes_with_nul(slice);
        if let Ok(c_str) = result {
            assert!(c_str.is_safe());
        }
    }
```

After some tests I think this above code should suffice, I tried doing extra validation by validating non-null within the string, but it seems to not make a difference.